common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1601,3 +1601,54 @@ def can_toggle_course_pacing(self):
           bool: False if the course has already started, True otherwise.
         """
         return datetime.now(UTC()) <= self.start
+
+
+class CourseSummary(object):
+    """
+    A lightweight course summary class, which constructs split/mongo course summary without loading
+    the course. It is used at cms for listing courses to global staff user.
+    """
+    default_display_name = "Empty"
+    course_info_fields = ['display_name', 'display_coursenumber', 'display_organization']
+
+    def __init__(self, course_locator, course_metadata):
+        """
+        Initialize and construct course summary
+        """
+        self.display_coursenumber = None
+        self.display_organization = None
+
+        self.course_metadata = course_metadata
+
+        self.id = course_locator  # pylint: disable=invalid-name
+        self.location = course_locator.make_usage_key('course', 'course')
+
+        # Extract & set course summary
+        for key, val in self.course_metadata.items():
+            setattr(self, key, val)
```

Comment:
Given that you're setting all of the values in `course_metadata` as attributes on this object, I think the api would be clearer if you acceptance `course_metadata` as `**course_metadata` (or `**kwargs`), so that you're matching the convention of `__init__` mapping to attributes that is kept by most data objects.